### DISCUSSION TITLE
How to close Testing Window in Rider using Actions instead of Keystrokes?

### DISCUSSION THREAD

**Discussion by Andras-Csanyi** [JetBrains/ideavim#976] [Q&A] [answered]
Hi all,

I'd like to achieve the following workflow:
- executing tests, I usually execute all of them, at once using a keystroke -> already solved
- closing the testing window

When I execute the tests (`map <leader>tt <Action>(RiderUnitTestRunSolutionAction)`) it opens the Testing window which is fine, I'd like to see the results. But, I cannot close it. When I check what Action is executed when the window is closed it is "HideActiveWindow", but this one requires to focus being in the window. If I start the testing using the vim command above the focus doesn't get there so the window cannot be closed. I checked what action is executed when the Testing window is opened by clicking on the Testing icon, the tracker says there is no recognisable action.

Using the Command-8 to close the testing window is a good alternative solution, but if it is possible I'd like to use vim like actions for this. The reason is that I'd like to use the same keystrokes in NeoVim, so the commands are the same in NeoVim and Rider/IntelliJ.

**Comment by citizenmatt** [comment, marked as answer]
Unfortunately, you can't use Vim-like shortcuts in a tool window. Since IdeaVim is an editor extension, its shortcuts only work in the editor. Once the tool window has the focus, you have to use an IDE shortcut.

**Reply by Andras-Csanyi** [reply]
Thanks for the answer! I believe I'll have further questions where the "it is an editor extension" will be the answer in one way or form. I really appreciate that you guys are answering within a few days.

**Reply by RNBermudez** [reply]
@citizenmatt would that also be true for popup dialogs such as `<GotoFile>` or `<FindUsages>`? Or is there a way to use ideavim to control the movement/accept the options here?

**Reply by citizenmatt** [reply]
It's also true for these windows too. Unless it's an editor, Vim shortcuts won't work. But you can always type to search/filter in lists and tree views, which is usually a very fast way to navigate.

**Comment by oscarlofwenhamn** [comment]
Hi,

It's possible to do a generic HideSideWindows action from the editor, which closes any 'Tool Window" but leaves the Explorer window open, for example. 

It's maybe not ideal, but that way it's possible to Esc focus back from the Tool Window to the editor and then HideSideWindows (trying out `<leader>vh` to see if that could be a good binding, but not yet convinced). 

Either way, it was a good find to offer a generic "please give my editor more room" action.
